src/test/java/de/retest/recheck/review/ignore/matcher/ElementIdMatcherTest.java

```diff
@@ -0,0 +1,83 @@
+package de.retest.recheck.review.ignore.matcher;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.descriptors.IdentifyingAttributes;
+
+public class ElementIdMatcherTest {
+
+	private ElementIdMatcher matcher;
+
+	@BeforeEach
+	void setUp() {
+		final Element element = mock( Element.class );
+		final IdentifyingAttributes attribs = mock( IdentifyingAttributes.class );
+		when( element.getIdentifyingAttributes() ).thenReturn( attribs );
+		when( attribs.get( "id" ) ).thenReturn( "someDiv" );
+		matcher = new ElementIdMatcher( element );
+	}
+
+	@Test
+	void should_match_when_id_is_parent() {
+		final Element childElement = mock( Element.class );
+		final IdentifyingAttributes childAttribs = mock( IdentifyingAttributes.class );
+		when( childElement.getIdentifyingAttributes() ).thenReturn( childAttribs );
+		when( childAttribs.get( "id" ) ).thenReturn( "someChildDiv" );
+
+		final Element element = mock( Element.class );
+		final IdentifyingAttributes attribs = mock( IdentifyingAttributes.class );
+		when( element.getIdentifyingAttributes() ).thenReturn( attribs );
+		when( attribs.get( "id" ) ).thenReturn( "someDiv" );
+
+		when( childElement.getParent() ).thenReturn( element );
+
+		assertThat( matcher ).accepts( childElement );
+	}
+
+	@Test
+	void should_match_when_id_is_parent_of_parent() {
```

Comment:
Do we need both tests? Parent and parent of parent?

Reply:
I would say yes. At one point i forgot the line `element = element.getParent();` which would lead to this test failing. I think we need the it to test all cases.